File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/kotlin/F16.P2.kt
fun probe(): Int {
    // BEGIN PROBE F16.P2
    val mp = mapOf("a" to 1)
    val total = mp.entries.sumOf { it.value }
    val miss = mp["b"] ?: 0
    return total + miss
    // END PROBE F16.P2
}

fun main() {
    println(probe())
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F16.P2.rs
use std::collections::HashMap;

fn probe() -> i32 {
// BEGIN PROBE F16.P2
let mp = HashMap::from([("a", 1)]);
let total: i32 = mp.values().sum();
let miss = *mp.get("b").unwrap_or(&0);
total + miss
// END PROBE F16.P2
}

fn main() {
    println!("{}", probe());
}
